Source/quests.cpp

```diff
@@ -337,8 +337,7 @@ void CheckQuests()
 	    && quest._qvar1 >= 2
 	    && (quest._qactive == QUEST_ACTIVE || quest._qactive == QUEST_DONE)
 	    && (quest._qvar2 == 0 || quest._qvar2 == 2)) {
-		// Move the quest trigger into world space, then spawn a portal at the same location
-		quest.position = quest.position.megaToWorld();

```

Comment:
will this still work if the game was started on vanilla?

Reply:
`quest.position` is initialized in `GenerateLevel` (in `drlg_l4.cpp`).
This logic runs every time a level is changed or loaded.
So in vanilla and in devilutionX this variable should be initialized every time.
Another evidence is that this transformation is only possible once but when returning from lazarus' lair it's executed every time.
That's why I came to the conclusion that it's safe to move `megaToWorld` to `GenerateLevel`. 🙂 
But I didn't test it vanilla.